packages/ckeditor5-dev-env/lib/utils/changelog.js

```diff
@@ -0,0 +1,249 @@
+
+		return `v${ packageJSON.version }`;
+	},
+
+	/**
+	 * Returns number of the next version.
+	 *
+	 * @param {String} currentVersion Current version in format "X.Y.Z".
+	 * @param {String} releaseType Type of the next release.
+	 * @returns {String}
+	 */
+	getNextVersion( currentVersion, releaseType ) {
+		if ( currentVersion.startsWith( 'v' ) ) {
+			currentVersion = currentVersion.slice( 1 );
+		}
+
+		const version = currentVersion.split( '.' ).map( ( n ) => parseInt( n ) );
+
+		if ( releaseType === 'major' ) {
+			return `${ version[ 0 ] + 1 }.0.0`;
+		} else if ( releaseType === 'minor' ) {
+			return `${ version[ 0 ] }.${ version[ 1 ] + 1 }.0`;
+		}
+
+		return `${ version[ 0 ] }.${ version[ 1 ] }.${ version[ 2 ] + 1 }`;
+	},
+
+	/**
+	 * Returns a level which represents a type of release based on the commits.
+	 *   - 0: major,
+	 *   - 1: minor,
+	 *   - 2: patch.
+	 *
+	 * An input array is a result of module {@link https://github.com/conventional-changelog/conventional-commits-parser}.
+	 *
+	 * @param {Array} commits
+	 * @returns {Number}
+	 */
+	getNewVersionType( commits ) {
+		let hasNewFeatures = false;
+
+		for ( const item of commits ) {
+			for ( const note of item.notes ) {
+				if ( note.title === 'BREAKING CHANGE' ) {
+					return 0;
+				}
+			}
+
+			if ( !hasNewFeatures && item.type === 'Feature' ) {
+				hasNewFeatures = true;
+			}
+		}
+
+		return hasNewFeatures ? 1 : 2;
+	},
+
+	/**
+	 * Parses command line arguments and returns them as a user-friendly hash.
+	 *
+	 * @returns {Object} options
+	 * @returns {String} options.token GitHub token used to authenticate.
+	 * @returns {Boolean} options.init Whether to create first release using this package.
+	 * @returns {Boolean} options.debug Whether to show additional logs.
+	 */
+	parseArguments( args = process.argv.slice( 2 ) ) {
+		const options = require( 'minimist' )( args, {
+			string: [
+				'token'
+			],
+
+			boolean: [
+				'init',
+				'debug'
+			],
+
+			default: {
+				init: false,
+				debug: false
+			}
+		} );
+
+		delete options._;
+
+		return options;
+	},
+
+	/**
+	 * Returns all changes between `currentTag` and `previousTag`.
+	 *
+	 * If `previousTag` is null, returns the whole file (first release).
+	 *
+	 * @param {String} currentTag
+	 * @param {String|null} previousTag
+	 * @returns {Promise}
+	 */
+	getLatestChangesFromChangelog( currentTag, previousTag ) {
+		currentTag = currentTag.replace( /^v/, '' );
+
+		return utils.getCurrentChangelog()
+			.then( ( changelog ) => {
+				changelog = changelog.replace( utils.changelogHeader, '' );
+
+				if ( previousTag ) {
+					previousTag = previousTag.replace( /^v/, '' );
+
+					changelog = changelog.match( new RegExp( `(## \\[${ currentTag }\\][\\w\\W]+)## \\[?${ previousTag }\\]?` ) )[ 1 ];
+				}
+
+				changelog = changelog.replace( new RegExp( `^## \\[?${ currentTag }\\]?.*` ), '' ).trim();
+
+				return Promise.resolve( changelog );
+			} );
+	},
+
+	/**
+	 * @returns {Promise}
```

Comment:
If the only thing this function does is reading a file from the system, please get rid of the promise. I've been writing this on and on in many PRs – use synchronous versions of Node.js methods.

PS. It took me 15s to try to understand what this function does. And I'm still unsure, which is exactly why KISS!